src/java.base/share/classes/sun/net/www/protocol/file/FileURLConnection.java

```diff
@@ -179,32 +174,28 @@ public long getLastModified() {
     public synchronized InputStream getInputStream()
         throws IOException {
 
-        int iconHeight;
-        int iconWidth;
-
         connect();
 
         if (is == null) {
             if (isDirectory) {
-                FileNameMap map = java.net.URLConnection.getFileNameMap();
 
                 StringBuilder sb = new StringBuilder();
 
-                if (files == null) {
-                    throw new FileNotFoundException(filename);
+                if (directoryListing == null) {
+                    throw new FileNotFoundException(file.getPath());
                 }
 
-                files.sort(Collator.getInstance());
+                directoryListing.sort(Collator.getInstance());
 
-                for (int i = 0 ; i < files.size() ; i++) {
-                    String fileName = files.get(i);
+                for (int i = 0; i < directoryListing.size() ; i++) {
+                    String fileName = directoryListing.get(i);
                     sb.append(fileName);
                     sb.append("\n");
```

Comment:
We can use a `StringJoiner` or stream to join the strings instead. (Note there is a suffix `\n` in the strings)

Reply:
`StringJoiner` will not include the trailing line break. I want to be conservative in this PR so reluctant to introduce streams for this.

Opted to use an enhanced for-loop. See f9efa3b.